### PR TITLE
Service definition changes for EntityValueResolver

### PR DESCRIPTION
**Pull request by eltharin**
This PR is related https://github.com/symfony/symfony/pull/57676 for add a variableLoader service to inject in EntityValueResolver to allow user to have it's own variables (eg: user) in mapEntity

### CODE REVIEW

File: src/DependencyInjection/DoctrineExtension.php [hunk 1]
```diff
@@ -533,6 +534,12 @@ protected function ormLoad(array $config, ContainerBuilder $container)
                     $controllerResolverDefaults['evict_cache'] ?? null,
                     $controllerResolverDefaults['disabled'] ?? false,
                 ]));
+            } else {
+                $container->getDefinition('doctrine.orm.entity_value_resolver')->setArgument(2, (new Definition(MapEntity::class)));
```
Comment (stof) [inline src/DependencyInjection/DoctrineExtension.php:540]: why this change ?

Comment (eltharin) [reply src/DependencyInjection/DoctrineExtension.php:540]: If not I can't set my new argument, I have this error : 

``
In DefinitionErrorExceptionPass.php line 48:
                                                                                                                                                      
  Invalid constructor argument 4 for service "doctrine.orm.entity_value_resolver": argument 3 must be defined before. Check your service definition. 
 ``

Comment (eltharin) [reply src/DependencyInjection/DoctrineExtension.php:540]: I can just remove the if to have only one initialisation.

**Comment by dmaicher** [comment]
Closing as the PR on Symfony side was also closed. See https://github.com/symfony/symfony/pull/57676#issuecomment-2243471133

**Comment by seb-jean** [comment]
Hello, what would be the solution to solve this problem?
